File: SurTerrain/src/Models/Clubcomp.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Models;

/**
 *
 * @author Hamidou
 */
public class Clubcomp {
    private int id_competition, id_club, idclubcomp;
    private String nom_club;

    public Clubcomp(int id_competition, int id_club, String nom_club) {
        this.id_competition = id_competition;
        this.id_club = id_club;
        this.nom_club = nom_club;
    }

    public Clubcomp(int id_competition, int id_club, int idclubcomp, String nom_club) {
        this.id_competition = id_competition;
        this.id_club = id_club;
        this.idclubcomp = idclubcomp;
        this.nom_club = nom_club;
    }

    public Clubcomp(String nom_club) {
        this.nom_club = nom_club;
    }
    

    public int getId_competition() {
        return id_competition;
    }

    public void setId_competition(int id_competition) {
        this.id_competition = id_competition;
    }

    public int getId_club() {
        return id_club;
    }

    public void setId_club(int id_club) {
        this.id_club = id_club;
    }

    public int getIdclubcomp() {
        return idclubcomp;
    }

    public void setIdclubcomp(int idclubcomp) {
        this.idclubcomp = idclubcomp;
    }

    public String getNom_club() {
        return nom_club;
    }

    public void setNom_club(String nom_club) {
        this.nom_club = nom_club;
    }

    @Override
    public String toString() {
        return "Clubcomp{" + "id_competition=" + id_competition + ", id_club=" + id_club + ", nom_club=" + nom_club + '}';
    }
    
    
}

File: src/Entity/Clubcomp.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Clubcomp
 *
 * @ORM\Table(name="clubcomp", indexes={@ORM\Index(name="fk_club", columns={"id_club"}), @ORM\Index(name="fk_comp", columns={"id_competition"})})
 * @ORM\Entity
 * @ORM\Entity(repositoryClass="App\Repository\ClubcompRepository")
 */
class Clubcomp
{
    /**
     * @var int
     *
     * @ORM\Column(name="idclubcomp", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idclubcomp;

    /**
     * @var string
     *
     * @ORM\Column(name="nom_club", type="string", length=20, nullable=false)
     */
    private $nomClub;

    /**
     * @var \Club
     *
     * @ORM\ManyToOne(targetEntity="Club")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="id_club", referencedColumnName="id_club")
     * })
     */
    private $idClub;

    /**
     * @var \Competition
     *
     * @ORM\ManyToOne(targetEntity="Competition")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="id_competition", referencedColumnName="id_competition")
     * })
     */
    private $idCompetition;

    public function getIdclubcomp(): ?int
    {
        return $this->idclubcomp;
    }

    public function getNomClub(): ?string
    {
        return $this->nomClub;
    }

    public function setNomClub(string $nomClub): self
    {
        $this->nomClub = $nomClub;

        return $this;
    }

    public function getIdClub(): ?Club
    {
        return $this->idClub;
    }

    public function setIdClub(?Club $idClub): self
    {
        $this->idClub = $idClub;

        return $this;
    }

    public function getIdCompetition(): ?Competition
    {
        return $this->idCompetition;
    }

    public function setIdCompetition(?Competition $idCompetition): self
    {
        $this->idCompetition = $idCompetition;

        return $this;
    }

    public function __toString()
    {
        return (string) $this->getIdclubcomp();
    }

}
